src/app/routes/getInitialData/index.test.js

```diff
@@ -77,8 +88,72 @@ describe('getInitialData', () => {
 
   describe('Rejected fetch', () => {
     it('should return an empty object', async () => {
-      const response = await callGetInitialData({}, mockFetchFailure);
-      expect(response).toEqual({});
+      const response = await callGetInitialData(
+        defaultContext,
+        mockFetchFailure,
+      );
+      expect(response).toEqual({
+        data: undefined,
+        isAmp: false,
+        service: 'news',
+        status: 502,
+      });
+    });
+  });
+
+  describe('Ares returns 200 status code, but invalid JSON', () => {
+    it('should return a 502 error code', async () => {
+      const response = await callGetInitialData(
+        defaultContext,
+        mockFetchInvalidJSON,
+      );
+
+      expect(response).toEqual({
+        data: undefined,
+        isAmp: false,
+        service: 'news',
+        status: 502,
+      });
+    });
+  });
+
+  describe('Ares returns a 404 status code', () => {
+    it('should return the status code as 404', async () => {
+      const response = await callGetInitialData(
+        defaultContext,
+        mockFetchNotFound,
+      );
+
+      expect(response).toEqual({
+        data: undefined,
+        isAmp: false,
+        service: 'news',
+        status: 404,
+      });
+    });
+  });
+
+  describe('Ares returns a non-200, non-404 status code', () => {
+    it('should log, and return the status code as 502', async () => {
+      global.console.warn = jest.fn();
```

Comment:
A nice to have, but we could have a test for `global.console.warn.totoHaveBeenCalledWith(...)` - would make turning this into logging much easier as the test would already exist